### PR TITLE
Prints cache memory usage in relevant stages

### PR DESCRIPTION
**Pull request by tinwelint**

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/MemoryUsageStatsProvider.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import org.neo4j.unsafe.impl.batchimport.stats.Stat;
+
+import static org.neo4j.helpers.Format.bytes;
+
+/**
+ * Provides {@link Stat statistics} about memory usage, as the key {@link Keys#memory_usage}
+ */
+public class MemoryUsageStatsProvider extends GenericStatsProvider implements Stat
+{
+    private final MemoryStatsVisitor.Home[] users;
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/MemoryUsageStatsProvider.java]: Why is this Visitable called "Home"?

Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/MemoryUsageStatsProvider.java]: Why are we using visitors for this?

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/RecordProcessorStep.java [hunk 2]
```diff
@@ -59,4 +66,11 @@ protected void done()
         super.done();
         processor.done();
     }
+
+    @Override
+    protected void addStatsProviders( Collection<StatsProvider> providers )
+    {
+        super.addStatsProviders( providers );
+        providers.addAll( additionalStatsProviders );
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/RecordProcessorStep.java]: Here you add the StatsProviders to super, but in the constructor you don't send them up to the parent class? Oh wait, this method adds StatsProviders _from_ this instance _to_ the given collection. This method has a confusing name. I suggest adding "To" or "Into" to the end of the name to make it more clear.

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/RecordProcessorStep.java [hunk 3]
```diff
@@ -31,13 +36,15 @@
...
+    private final Collection<StatsProvider> additionalStatsProviders;
 
     public RecordProcessorStep( StageControl control, String name, int workAheadSize, int movingAverageSize,
-            RecordProcessor<T> processor, boolean endOfLine )
+            RecordProcessor<T> processor, boolean endOfLine, StatsProvider... additionalStatsProviders )
     {
         super( control, name, workAheadSize, movingAverageSize, 1 );
         this.processor = processor;
         this.endOfLine = endOfLine;
+        this.additionalStatsProviders = asList( additionalStatsProviders );
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/RecordProcessorStep.java]: Why are these not sent up through super()?

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/RecordProcessorStep.java (hunk 2 above)
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/RecordProcessorStep.java]: You could alternatively call it "collectStatsProviders"

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/MemoryUsageStatsProvider.java (hunk 1 above)
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/MemoryUsageStatsProvider.java]: I found it cute, a home to visit. Although the method name in `Home` should be `accept`. Better suggestions, `Visitable` ?

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/RecordProcessorStep.java (hunk 3 above)
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/RecordProcessorStep.java]: Forgot this place when changing the previously temporary implementation. I'd guess I was a little too quick yesterday evening

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/RecordProcessorStep.java (hunk 2 above)
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/RecordProcessorStep.java]: yes I like that more

**Pull request merged**